mpas_analysis/shared/plot/plotting.py

```diff
@@ -827,27 +887,37 @@ def setup_colormap(config, configSectionName, suffix=''):
 
     Authors
     -------
-    Xylar Asay-Davis, Milena Veneziani
+    Xylar Asay-Davis, Milena Veneziani, Greg Streletz
     '''
 
     colormap = plt.get_cmap(config.get(configSectionName,
                                        'colormapName{}'.format(suffix)))
+
     indices = config.getExpression(configSectionName,
-                                   'colormapIndices{}'.format(suffix))
+                                   'colormapIndices{}'.format(suffix),
+                                   usenumpyfunc=True)
+
     colorbarLevels = config.getExpression(configSectionName,
-                                          'colorbarLevels{}'.format(suffix))
-
-    # set under/over values based on the first/last indices in the colormap
-    underColor = colormap(indices[0])
-    overColor = colormap(indices[-1])
-    if len(colorbarLevels)+1 == len(indices):
-        # we have 2 extra values for the under/over so make the colormap
-        # without these values
-        indices = indices[1:-1]
-    colormap = cols.ListedColormap(colormap(indices),
-                                   'colormapName{}'.format(suffix))
-    colormap.set_under(underColor)
-    colormap.set_over(overColor)
+                                          'colorbarLevels{}'.format(suffix),
+                                          usenumpyfunc=True)
+
+    if colorbarLevels is not None:
```

Comment:
I don't think `config.getExpression` ever returns `None` anymore.  (It used to when an option was absent.)  Instead, an exception will be raised if there is not an option with that name.  If you want this code to return `None` for the `colorbarLevels` if there aren't any in this section of the config file, you will need a `try`/`exept` around the `config.getExpression` where you set colorbarLevels to `None` if the appropriate exception has been raised (`ConfigParser.NoOptionError`).